promql/engine.go

```diff
@@ -782,11 +791,14 @@ func (ev *evaluator) recover(ws *storage.Warnings, errp *error) {
 	}
 }
 
-func (ev *evaluator) Eval(expr parser.Expr) (v parser.Value, ws storage.Warnings, err error) {
+func (ev *evaluator) Eval(expr parser.Expr) (val parser.Value, ws storage.Warnings, err error) {
 	defer ev.recover(&ws, &err)
 
-	v, ws = ev.eval(expr)
-	return v, ws, nil
+	val, ws = ev.eval(expr)
+	if mat, ok := val.(Matrix); ok {
+		ev.samplesStats.Increment(mat.TotalSamples())
```

Comment:
This deserves a comment, that's we're doing a single count of the overall output.

Reply:
This comment reads oddly. Something like Include output samples in overall samples touched would be clearer.